pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -167,9 +166,18 @@ type Compactor struct {
 	// Subservices manager.
 	subservices        *services.Manager
 	subservicesWatcher *services.FailureWatcher
+
+	// sharded by objectType
+	containers map[string]storeContainer
+}
+
+type storeContainer struct {
+	tableMarker        retention.TableMarker
+	sweeper            *retention.Sweeper
+	indexStorageClient shipper_storage.Client
 }
 
-func NewCompactor(cfg Config, objectClient client.ObjectClient, schemaConfig config.SchemaConfig, limits *validation.Overrides, r prometheus.Registerer) (*Compactor, error) {
+func NewCompactor(cfg Config, objectClients map[string]client.ObjectClient, schemaConfig config.SchemaConfig, limits *validation.Overrides, r prometheus.Registerer) (*Compactor, error) {
```

Comment:
Nit: I'm finding the use of the term "object" in this PR to be a bit confusing; these are object _store_ clients - so I think the more clear name would be `storeClients`, `storeType`, etc.

Reply:
agree, I'll do the needful